### PR TITLE
[Enhancement] Support getUUID of external database & table

### PR DESCRIPTION
**Pull request by letian-jiang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

To support RBAC, we should assign the UUID of every database and table. 
* database.getUUID -> catalog.db
* table.getUUID -> catalog.db.table.createTime

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 20 (55.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergRESTCatalog.java|0|1|00.00%|[79]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|0|1|00.00%|[156]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|1|00.00%|[139]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergHiveCatalog.java|0|1|00.00%|[174]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergGlueCatalog.java|0|1|00.00%|[166]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|1|00.00%|[408]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|10|11|90.91%|[322]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 11 / 20 (55.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergRESTCatalog.java|0|1|00.00%|[79]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|0|1|00.00%|[156]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|1|00.00%|[139]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergHiveCatalog.java|0|1|00.00%|[174]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergGlueCatalog.java|0|1|00.00%|[166]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|1|00.00%|[408]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|10|11|90.91%|[321]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 20 (65.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergRESTCatalog.java|0|1|00.00%|[79]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|1|00.00%|[139]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergHiveCatalog.java|0|1|00.00%|[174]|
|:large_blue_circle:|com/starrocks/connector/iceberg/glue/IcebergGlueCatalog.java|0|1|00.00%|[166]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/connector/iceberg/IcebergUtil.java|0|1|00.00%|[408]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|11|11|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/connector/hive/HiveMetastoreApiConverter.java|1|1|100.00%|[]|

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

File: fe/fe-core/src/test/java/com/starrocks/catalog/IcebergTableTest.java [hunk 1]
```diff
@@ -49,6 +50,7 @@ public class IcebergTableTest {
 
     @Before
     public void setUp() {
+        catalogName = "catalog0";
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/catalog/IcebergTableTest.java:53]: this table for test is created manually. there is no need to test it. catalog must be empty

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 13 / 17 (76.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|3|4|75.00%|[248]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[772]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|1|1|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/catalog/IcebergTableTest.java (hunk 1 above)
Comment (letian-jiang) [reply fe/fe-core/src/test/java/com/starrocks/catalog/IcebergTableTest.java]: removed

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/catalog/Database.java [hunk 2]
```diff
@@ -305,6 +309,9 @@ public long getId() {
      * @return unique id of database in string format
      */
     public String getUUID() {
+        if (CatalogMgr.isExternalCatalog(catalogName)) {
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:312]: why not abstact something like `ExternalDatabase()` which extends from `Database` and overrides `getUUID()`?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 10 / 14 (71.43%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|0|1|00.00%|[157]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|1|00.00%|[139]|
|:large_blue_circle:|com/starrocks/catalog/DeltaLakeTable.java|0|1|00.00%|[83]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|4|5|80.00%|[772]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/MetadataMgr.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/CatalogMgr.java|1|1|100.00%|[]|

**Review by nshangyiming — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Youngwb — APPROVED** [review]

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
starrocks_admit_test

**Comment by stephen-shelby** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
run starrocks_admit_test

**Comment by letian-jiang** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**